### PR TITLE
Feature/457 newsfeed filters cherry pick

### PR DESCRIPTION
**Pull request by chao-mu**
Add category based newsfeed filters

### CODE REVIEW

**Comment by chao-mu** [comment]
References  #457

File: frontend/src/newsfeed/list/NewsfeedList.tsx [hunk 1]
```diff
@@ -152,6 +193,28 @@ const NewsfeedList: React.FC<NewsfeedListProps> = ({
...
+            finalFilters = proposedFilters.filter(
+                (filter) => filter != AllCategoriesFilterName,
+            );
+        }
+
+        setFilters(finalFilters);
+    };
+
+    const allEntries = data?.entries ?? [];
+    const shownEntries = allEntries.filter((entry) =>
```
Comment (jackstenglein) [inline frontend/src/newsfeed/list/NewsfeedList.tsx:214]: Can we check that showAdditionalFilters is true before performing this operation? If it's false, then shownEntries will always equal allEntries and there isn't any reason to loop through all the entries.

**Comment by chao-mu** [comment]
Problems addressed.

Added an eslint plugin that supports autofix in more cases with loose vs. strict equality. eslint doesn't fix all the cases out of the box in order to support legacy code.

**Comment by chao-mu** [comment]
Fixes #346
Fixes #457

**Pull request merged**